src/components/InputTag/InputTag.tsx

```diff
@@ -1,16 +1,44 @@
-import { DetailedHTMLProps, InputHTMLAttributes } from "react"
-import style from "./InputTag.module.scss"
+import { DetailedHTMLProps, InputHTMLAttributes } from "react";
+import style from "./InputTag.module.scss";
+import { isNumberValid } from "./validation";
 
 type Props = {
-    label: string,
-    isOn: boolean,
-} & DetailedHTMLProps<InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>
+    id: string;
+    disabled: boolean;
+    onChange: (state: number) => void;
+} & Omit<
+    DetailedHTMLProps<InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>,
+    "onChange" | "disabled"
+>;
 
-export function InputTag({label, isOn, ...inputProps}: Props) {
+const onChangeInput = (
```

Comment:
Let the parent handle checking wheter the value is correct or not rather than doing it inside the InputTag